compio-runtime/src/runtime/mod.rs

```diff
@@ -88,7 +97,15 @@ impl RuntimeInner {
 
     pub fn run(&self) {
         loop {
-            let next_task = self.runnables.pop();
+            let next_task = self.local_runnables.borrow_mut().pop_front();
+            if let Some(task) = next_task {
+                task.run();
+            } else {
+                break;
+            }
+        }
+        loop {
```

Comment:
Separated loop for local and global queue may starve global task when local queue is busy.